docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,209 @@
+given Context = ctx
+```
+
+You can find an extensive discussion on the above change in its [PR #10538](https://github.com/lampepfl/dotty/pull/10538).
+
+## Drop `as` in patterns
+Since we dropped `as` from `given`s, we lost a strong reason for having `as` at all. For this reason, we dropped them also from patterns. If previously the correct binding syntax is patterns was:
+
+```scala
+case opt as Some(foo)
+```
+
+Now it got reverted to the Scala 2 style:
+
+```scala
+case opt @ Some(foo)
+```
+
+## Switch back to the old context function closure syntax
+Previously, context function closures were written as follows:
+
+```scala
+(using s: Show[String]) => s.show("foobar")
+```
+
+Now, we have reverted to the following syntax:
+```scala
+(s: Show[String]) ?=> s.show("foobar")
+```
+
+# `Matchable` trait – a new top-level type
+The motivation for this change is the fact that it is possible to access the underlying type referenced by an opaque type using pattern matching:
+
+```scala
+val imm = IArray(1,2,3) // supposedly immutable...
+imm match
+   case a: Array[Int] => a(0) = 0  // but that's shown to be lie
+```
+
+To address this change, a new trait, `Matchable`, is introduced at the top of the type hierarchy:
+
+```scala
+abstract class Any:
+  def asInstanceOf
+  def ==
+  def !=
+  def ##
+  def equals
+  def hashCode
+  def toString
+
+trait Matchable extends Any:
+  def isInstanceOf
+  def getClass
+
+class AnyVal extends Any, Matchable
+
+class Object extends Any, Matchable
+```
+
+Matchable is currently a marker trait without any methods. Over time we might migrate methods getClass and isInstanceOf to it, since these are closely related to pattern-matching.
+
+With the addition of this type, we now make the compiler emit warnings when trying to do a trick like the one discussed above:
+
+```scala
+-- Warning: i7314.scala:6:12 ---------------------------------------------------
+6 |    case a: Array[Int] =>
+  |            ^^^^^^^^^^
+  |            pattern selector should be an instance of Matchable,
+  |            but it has unmatchable type opaques.IArray[Int] instead
+```
+
+The change involved an extensive discussion which you can read in the [PR #10670](https://github.com/lampepfl/dotty/pull/10670). Read more about this change in the [documentation](https://dotty.epfl.ch/docs/reference/other-new-features/matchable.html).
+
+# Tooling improvements
+As the work on Scala 3 gets closer to the stable release, the focus increasingly shifts on the tooling available to get started with Scala 3.
+
+For a while now, we are not using the old dottydoc documentation tool for building the documentation. We are developing an entirely new tool, scala3doc, from scratch.
+
+As part of the tooling effort, this new Scala 3 documentation tool is rapidly improved. [PR #10522](https://github.com/lampepfl/dotty/pull/10522) proves that the doctool can generate documentation for the community build projects.
```

Comment:
```suggestion
As part of the tooling effort, we are rapidly improving the new Scala 3 documentation tool, `scala3doc`. As shown in the [PR #10522](https://github.com/lampepfl/dotty/pull/10522), it can now generate documentation for all the community build projects.
```

Reply:
We should stress here explicitly that the old tool is no longer in use.